mono/mini/interp/transform.c

```diff
@@ -887,6 +904,27 @@ jit_call_supported (MonoMethod *method, MonoMethodSignature *sig)
 	return FALSE;
 }
 
+#ifdef ENABLE_EXPERIMENT_TIERED
+static gboolean
+jit_call2_supported (MonoMethod *method, MonoMethodSignature *sig)
+{
+	if (sig->param_count > 6)
+		return FALSE;
+	if (sig->pinvoke)
+		return FALSE;
+	if (method->flags & METHOD_ATTRIBUTE_PINVOKE_IMPL)
+		return FALSE;
+	if (method->iflags & METHOD_IMPL_ATTRIBUTE_INTERNAL_CALL)
+		return FALSE;
+	if (method->is_inflated)
+		return FALSE;
+	if (method->string_ctor)
+		return FALSE;
+
+	return TRUE;
+}
```

Comment:
```
return !(sig->param_count > 6 ||
  sig->pinvoke ||
  (method->flags & METHOD_ATTRIBUTE_PINVOKE_IMPL) ||
  (method->iflags & METHOD_IMPL_ATTRIBUTE_INTERNAL_CALL) ||
  method->is_inflated ||
  method->string_ctor);
```
?

Reply:
I find the separate `if`s much easier to read.

Reply:
```
return sig->param_count <= 6  &&
  !sig->pinvoke &&
  !(method->flags & METHOD_ATTRIBUTE_PINVOKE_IMPL)  &&
  !(method->iflags & METHOD_IMPL_ATTRIBUTE_INTERNAL_CALL)  &&
  !method->is_inflated  &&
  !method->string_ctor;
```
?
How is the equivalence not obvious?

Reply:
it's using the same style as the function above (and it's also more readable imho).

Reply:
Simple boolean expressions are well within the abilities to understand by mono developers, and neighboring practise should not be the only consideration.